integration-tests/src/test/java/org/alfresco/core/NodeApiServiceIntegrationTest.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2021-2021 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.alfresco.core;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import java.io.File;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import org.alfresco.core.handler.NodesApiClient;
+import org.alfresco.core.model.NodeBodyCopy;
+import org.alfresco.core.model.NodeBodyCreate;
+import org.alfresco.core.model.NodeChildAssociationPaging;
+import org.alfresco.core.model.NodeEntry;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.testcontainers.containers.DockerComposeContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.shaded.org.apache.commons.lang.RandomStringUtils;
+
+@Testcontainers
+@SpringBootTest
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class NodeApiServiceIntegrationTest {
+
+    @Container
+    public static DockerComposeContainer compose = new DockerComposeContainer(new File("src/test/resources/docker-compose.yml"))
+        .withLocalCompose(true)
+        .withExposedService("alfresco", 8080, Wait.forListeningPort().withStartupTimeout(Duration.ofSeconds(90)));
+
+    @Autowired
+    private NodesApiClient nodesApiClient;
+
+    public static String getRandomNodeName() {
+        return "test-node-" + RandomStringUtils.randomNumeric(8);
+    }
+
+    @BeforeAll
+    public void setUp() {
+        long awaitTimeoutInMilliseconds = 90000;
+        await()
+            .atMost(awaitTimeoutInMilliseconds, TimeUnit.MILLISECONDS)
+            .pollDelay(2000, TimeUnit.MILLISECONDS)
+            .untilAsserted(() ->
+                assertThat(nodesApiClient
+                    .getNode("-my-", null, null, null)
+                    .getStatusCode()).isEqualTo(HttpStatus.OK));
+    }
+
+    @Test
+    void should_createNode_and_getNode() {
```

Comment:
@zoltanpalfi are we finally OK with this naming?. I.e. When we're only testing the happy path, we don't use "_when_" and the test outcome is the name of the method we're testing. If so, should we add this to our code convention wiki?